### PR TITLE
Allow ESM module for custom middlewares

### PR DESCRIPTION
**Pull request by 5ika**
### What does it do?

Fix issue #14146 

### Why is it needed?

With TS version of Strapi, we can use ESM module imports but we can't import custom middleware in `.ts` module file.

### How to test it?

See https://github.com/strapi/strapi/issues/14146

### Related issue(s)/PR(s)

#14146

### CODE REVIEW

**Review by Convly — CHANGES_REQUESTED** [review]
Hey, thank you for taking the time to open a PR,

If we want to enable loading ESM here, we could use the `import-default` util from [here](https://github.com/strapi/strapi/blob/features/api-token-v2/packages/core/strapi/lib/utils/import-default.js#L1) and move it to the `@strapi/utils` package, so that we're consistent with the rest of the logic where we allow using ESMs.

**Comment by 5ika** [comment]
@Convly Thanks ! I've updated the code to use import-default function.
I'm not sure I understand the last part of your comment: you mean move the function to @strapi/utils then update all occurrences in the code base ?

**Comment by Convly** [comment]
> @Convly Thanks ! I've updated the code to use import-default function. I'm not sure I understand the last part of your comment: you mean move the function to @strapi/utils then update all occurrences in the code base ?

Hey, really sorry for the late answer, it got lost in the Github notifications.
That's I was suggesting indeed, we could move the content of [this file](https://github.com/strapi/strapi/blob/main/packages/core/strapi/lib/utils/import-default.js) to the [strapi/utils](https://github.com/strapi/strapi/tree/main/packages/core/utils/lib) package.

**Comment by 5ika** [comment]
> Hey, really sorry for the late answer, it got lost in the Github notifications. That's what I was suggesting indeed, we could move the content of [this file](https://github.com/strapi/strapi/blob/main/packages/core/strapi/lib/utils/import-default.js) to the [strapi/utils](https://github.com/strapi/strapi/tree/main/packages/core/utils/lib) package.

@Convly No problem for the delay and thank you for your response. I've moved import-default to strapi/utils package as asked :+1:

File: packages/core/strapi/lib/core/app-configuration/load-config-file.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const path = require('path');
 const fs = require('fs');
 const { templateConfiguration, env } = require('@strapi/utils');
```
Comment (Convly) [inline packages/core/strapi/lib/core/app-configuration/load-config-file.js:5]:
```suggestion
const { templateConfiguration, env, importDefault } = require('@strapi/utils');
```

File: packages/core/strapi/lib/services/server/middleware.js [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 const path = require('path');
 const { propOr, isArray, isNil } = require('lodash/fp');
+const importDefault = require('@strapi/utils/lib/import-default');
```
Comment (Convly) [inline packages/core/strapi/lib/services/server/middleware.js:5]:
```suggestion
const { importDefault } = require('@strapi/utils');
```

**Review by Convly — CHANGES_REQUESTED** [review]
Thanks, I'm afraid those files would break if we don't change the importDefault require
- /packages/core/strapi/lib/core/loaders/apis.js
- /packages/core/strapi/lib/core/loaders/middlewares.js
- /packages/core/strapi/lib/core/loaders/policies.js
- /packages/core/strapi/lib/core/loaders/src-index.js

**Comment by 5ika** [comment]
> Thanks, I'm afraid those files would break if we don't change the importDefault require
> 
>     * /packages/core/strapi/lib/core/loaders/apis.js
> 
>     * /packages/core/strapi/lib/core/loaders/middlewares.js
> 
>     * /packages/core/strapi/lib/core/loaders/policies.js
> 
>     * /packages/core/strapi/lib/core/loaders/src-index.js

Oh, sorry. I thought I've updated all occurrences. It's now updated.
Thank you

**Review by Convly — APPROVED** [review]
LGTM
Thanks for your contribution :rocket:

**Comment by Convly** [comment]
_(I'll just do a last round of verification (probably today or tomorrow) before merging)_

**Pull request merged**